src/client/apiClient.ts

```diff
@@ -0,0 +1,95 @@
+import { OutgoingHttpHeaders } from "http";
+
+export interface ApiClient {
+  startSimulation: (simulationId: string, options?: StartSimulationRequest) => Promise<StartSimulationResponse>;
+  getRunInformation: (runId: string) => Promise<RunInformationResponse>;
+  getSimulations: () => Promise<SimulationResponse[]>;
+  abortRun: (runId: string) => Promise<void>;
+  getUserConcurrentMetric: (runId: string, scenario: string) => Promise<SeriesResponse[]>;
+  getRequestsSummary: (runId: string) => Promise<RequestsSummaryResponse>;
+}
+
+export const apiClient = (): ApiClient => {
+  const client = new http.HttpClient();
+  const conf: HttpConf = {
+    // TODO RND-5 configurable
+    baseUrl: "https://cloud.gatling.io/api/public",
+    apiToken: ""
+  };
+  return {
+    startSimulation: (simulationId, options) => {
+      const payload = options ? options : {};
+      return postJson(client, conf, "/simulations/start", payload, { simulation: simulationId });
+    },
+    getRunInformation: (runId) => getJson(client, conf, "/run", { run: runId }),
+    getSimulations: () => getJson(client, conf, "/simulations"),
+    abortRun: (runId) => postJson(client, conf, "/simulations/abort", {}, { run: runId }),
+    getUserConcurrentMetric: (runId, scenario) => getJson(client, conf, "/series", seriesParams(runId, scenario)),
+    getRequestsSummary: (runId) => getJson(client, conf, "/summaries/requests", { run: runId })
+  };
+};
+
+interface HttpConf {
+  baseUrl: string;
+  apiToken: string;
+}
+
+const seriesParams = (runId: string, scenario: string) => ({
+  run: runId,
+  scenario: scenario,
+  group: "",
+  request: "",
+  node: "",
+  remote: "",
+  metric: "usrActive"
+});
+
+const getJson = async <T>(
+  client: http.HttpClient,
+  conf: HttpConf,
+  path: string,
+  params?: Record<string, string>
+): Promise<T> => {
+  const response = await client.getJson<T>(buildUrl(conf, path, params), headers(conf));
+  return handleJsonResponse(response);
+};
+
+const postJson = async <T>(
+  client: http.HttpClient,
+  conf: HttpConf,
+  path: string,
+  payload: any,
+  params?: Record<string, string>
+): Promise<T> => {
+  const response = await client.postJson<T>(buildUrl(conf, path, params), payload, headers(conf));
+  return handleJsonResponse(response);
+};
+
+const headers = (conf: HttpConf): OutgoingHttpHeaders => ({ Authorization: conf.apiToken });
+
+const buildUrl = (conf: HttpConf, path: string, queryParams?: Record<string, string>): string => {
+  const resourceUrl = conf.baseUrl + path;
+  const url = new URL(resourceUrl);
+  if (queryParams) {
+    for (const key in queryParams) {
+      url.searchParams.append(key, queryParams[key]);
+    }
+  }
+  return url.toString();
+};
+
+const handleJsonResponse = <T>(response: TypedResponse<T>): T => {
+  if (response.statusCode === HttpCodes.NotFound || response.result === null) {
+    throw new HttpClientError("Unexpected empty response", HttpCodes.NotFound);
+  }
+  return response.result;
+};
```

Comment:
Question: shouldn't we exclude others non ok responses? Are they already rejected by default?

Reply:
From the doc for `HttpClient.getJson()`:

> Be aware that not found returns a null. Other errors (4xx, 5xx) reject the promise

In practice these functions reject the promise with an `HttpClientError`, you can see an example in the test called "startSimulation unauthorized error".

Note that it is only true for the `getJson`/`postJson`/`putJson`/etc. functions - if you use the base `get`/`post`/`put`/etc. functions, you need to handle error statuses yourself.